documentation/decisions/0007-model-resolution.md

```diff
@@ -0,0 +1,102 @@
+  triple store.
+* Definition 7: A `namespace` is the logical grouping of named model elements that share the same
+  namespace identifier.
+* Definition 8: A `namespace element definition` is the pseudo-model element definition statement
+  `?namespace rdf:type samm:Namespace` where `?namespace` is a namespace identifier and additional
+  optional statements with the same subject and RDF properties `samm:preferredName`,
+  `samm:description` and `samm:see`. The namespace element definition is contained in an aspect
+  model file, but it is not logically contained in a namespace.
+* Definition 9: The set of aspect model files that contain model elements that all share the same
+  namespace is called the _namespace files_. I.e., all model elements from a set of namespace files
+  are all defined in the same namespace.
+* Definition 10: A _models root_ is an identifier for a place where aspect model files are stored.
+  It could be a file system folder, a URL of a GitHub repository, a SPARQL endpoint URL or similar.
+* Definition 11: A _resolution strategy_ is a function that takes a models root and an Aspect Model
+  Element Identifier as input and returns the model element definition for the model element
+  identified by the Aspect Model Element Identifier, including information about the aspect model
+  file that contains the model element definition; or indicates a resolution failure (i.e., no model
+  element definition exists in this models root). It is allowed that the resolution strategy can be
+  configured with additional configuration (such as credentials), but this configuration should not
+  be required to be given in each function call.
+* Definition 12: An _aspect model resolver_ is a function that takes at least one models root, at
+  least one resolution strategy and at least one Aspect Model Element Identifier as input and
+  returns model element definitions for all transitively referenced model elements. Note that model
+  elements referred to by samm:see are not considered referenced in an aspect model resolver.
+
+### Restrictions - Rules for files, namespaces and model elements
+
+* An aspect model file may only contain model elements of exactly one namespace.
+* Model element definitions of a namespace may be spread across any number of aspect model files
+  (i.e., the namespace's namespace files).
+* A model element definition may only appear once per namespace and aspect model file.
+* A namespace may contain any number of model elements, including `samm:Aspect`.
+* An aspect model file may contain at most one model element definition for samm:Aspect.
+* A namespace's set of namespace files may contain at most one namespace element definition, which must match the namespace.
+* In filesystem-based model storage, aspect model files must end with ".ttl".
+* In filesystem-based model storage, models should be stored in the directory structure
+  _models_root_/_package_part_/_version_part_/AspectModelFile.ttl.
```

Comment:
- is "package part" now a new definition of the part of the namespace without version (in reverse domain notation)? Make it a first class definition? 
- Should version or namespace version also be a separate definition

Reply:
Added definitions for
* _namespace main part_ (instead of "package part")
* _namespace version part_